### ISSUE TITLE
Forcing preference order for decorators (to make parametrize() work for mpl_image_compare())

### ISSUE THREAD

**Issue by bsipocz** [matplotlib/pytest-mpl#15]
In photutils we have tests that use ``@pytest.mark.parametrize()`` to loop through the different aperture classes. However using ``@pytest.mark.mpl_image_compare()`` in this framework doesn't work as it would be ideal as the first decorator doesn't seem to paramtrize the arguments of the latter.

This is what I tried to do:
```
@pytest.mark.skipif('not HAS_MATPLOTLIB')
@pytest.mark.parametrize(('aperture_class', 'params'), TEST_APERTURES)
@pytest.mark.mpl_image_compare(filename='{0}.png'.format(aperture_class), tolerance=1.5)
def test_aperture_plots(aperture_class, params):
```

and ended up with a ``NameError``:

```
==================================== ERRORS ====================================
_________ ERROR collecting photutils/tests/test_aperture_photometry.py _________
photutils/tests/test_aperture_photometry.py:66: in <module>
>   @pytest.mark.mpl_image_compare(filename='{0}.png'.format(aperture_class), tolerance=1.5)
E       NameError: name 'aperture_class' is not defined
```

 @cdeil mentioned in https://github.com/astropy/astroplan/issues/86#issuecomment-136130398 that it may be possible to configure this in pytest, however I haven't had the time to check it out.

**Comment by astrofrog** [comment]
I don't think you'll be able to pass the aperture_class variable but I think what we *can* do is make the mpl_image_compare decorator name the images based on the parameters.

**Comment by astrofrog** [comment]
PR forthcoming

**Comment by astrofrog** [comment]
Fixed in #24 (feel free to re-open if that isn't enough)
